src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
 
+        Object.assign(state, extraData);
+        break;
+      case 'removeProperties':
+        const { keysToRemove } = action;
+
+        for (const keyToRemove of keysToRemove) {
+          if (keyToRemove in state) {
+            delete state[keyToRemove];
+          }
+        }
+        break;
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+      default:
+    }
+  }
+}
```

Comment:
function should return state

Reply:
@choeqq In this particular example I would have to reassign value to the state
`state = { ...state, ...action.extraData };`
which I cannot do with function parameter, so Object.assign is the best option here.